code/modules/mob/living/carbon/human/zombie.dm

```diff
@@ -133,7 +133,10 @@
 		return
 	if(!claw.zombium_per_hit)
 		return
-	reagents.add_reagent(/datum/reagent/zombium, modify_by_armor(claw.zombium_per_hit, BIO, 0, zombie.get_limbzone_target()))
+	if(isrobot(src) || issynth(src))
```

Comment:
Don't use these istypes, just check the species flags. NO_CHEM_METABOLIZATION is probably the right one.